src/server/rdb_save.cc

```diff
@@ -666,7 +666,14 @@ error_code RdbSerializer::SaveStreamConsumers(streamCG* cg) {
 }
 
 error_code RdbSerializer::SendFullSyncCut() {
-  return WriteOpcode(RDB_OPCODE_FULLSYNC_END);
+  RETURN_ON_ERR(WriteOpcode(RDB_OPCODE_FULLSYNC_END));
+
+  // RDB_OPCODE_FULLSYNC_END followed by 8 bytes of 0.
+  // The reason for this is that some opcodes require to have at least 8 bytes of data
+  // in the read buffer (EnsureRead) and since RDB_OPCODE_FULLSYNC_END is one of
```

Comment:
some opcodes are expected to have at least 8 bytes of data in the read buffer when consuming the rdb data

Reply:
?

Reply:
done